test/aievec/conv2d_f32_unaligned_loads.mlir

```diff
@@ -0,0 +1,100 @@
+// This test case will directly return the result generated from -affine-super-vectorize
+// because in transfer_read %arg2[%arg3, %arg4], dim 1's memref shape size(2046) cannot be
+// divisible by the vector lanes(8).
+
+// RUN: aie-opt %s --affine-loop-unroll="unroll-full unroll-full-threshold=3" --canonicalize -affine-super-vectorize="virtual-vector-size=8" --aie-vectorize -split-input-file -unaligned-loads-check=true --debug-only=aie-vect 2>&1 | FileCheck %s
```

Comment:
This is a pretty complex run line... It might be better to have a focused unit test that just looks at one aspect... i.e. start with a minimal loop nest and a single load/store and show that it's not handled?

Reply:
Also, if this is a negative test, it should probably check for the message as output?

Reply:
> This is a pretty complex run line... It might be better to have a focused unit test that just looks at one aspect... i.e. start with a minimal loop nest and a single load/store and show that it's not handled?

I reuse the test conv2d_f32.mlir to show this is an unaligned lit test when we turn -unaligned-loads-check to true. I can write a small test case by your request.

Reply:
> Also, if this is a negative test, it should probably check for the message as output?

The debug message check is at line 27.

Reply:
Test conv2d_f32_unaligned_load duplicates much of conv2d_f32. The conv2d_f32_unaligned_load should not check the MLIR that is produced but instead just check that the error message was emitted.